yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -109,7 +110,14 @@ class ExecutorRunnable(
     }
 
     // Send the start request to the ContainerManager
-    nmClient.startContainer(container, ctx)
+    try {
+      nmClient.startContainer(container, ctx)
+    } catch {
+      case ex: YarnException =>
+        logError("Exception while start container %s on host %s:"
```

Comment:
The same exception shouldn't be logged twice.  If we're going to rethrow it, we should wrap it in another exception and include the additional info in that exception's message.